packages/hash/src/index.ts

```diff
@@ -2,62 +2,93 @@
 // Inspired by https://github.com/garycourt/murmurhash-js
 // Ported from https://github.com/aappleby/smhasher/blob/61a0530f28277f2e850bfc39600ce61d02b518de/src/MurmurHash2.cpp#L37-L86
 
-export default function murmur2(str: string): string {
+const hasTextEncoder = typeof TextEncoder !== 'undefined'
+
+const encoder = hasTextEncoder ? new TextEncoder() : undefined
+
+let bufferLength = -1
+let buffer: ArrayBuffer
+let uint8View: Uint8Array
+let int32View: Int32Array
+
+function encode(input: string): number {
+  // Legacy IE11 support
+  if (hasTextEncoder === false) {
+    const bytes = []
+    for (let i = 0; i < input.length; i++) {
+      const codePoint = input.charCodeAt(i)
+      if (codePoint > 0xff) {
+        bytes.push(codePoint >>> 8)
+        bytes.push(codePoint & 0xff)
+      } else {
+        bytes.push(codePoint)
+      }
+    }
+    uint8View = new Uint8Array(bytes)
+    int32View = new Int32Array(uint8View.buffer, 0, Math.floor(bytes.length / 4))
+
+    return bytes.length
+  }
```

Comment:
This needs to encode like `TextEncoder` since the hash needs to return the same results if there e.g. is `TextEncoder` when a page is server-rendered but not on the client

Reply:
Self notes:
[utf16 to unicode](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/charCodeAt)
[unicode to utf8](https://gist.github.com/MightyPork/52eda3e5677b4b03524e40c9f0ab1da5)